### PR TITLE
Rotate some OG 3D pickups to make more sense

### LINKED ISSUE #253: Feature: Rotate 3D pickups
The four 3D pickups surrounding the obelisk are facing the same direction, would it be possible to rotate these so that all four pickups are facing outwards from the obelisk towards the respective drawbridges?

### PR DESCRIPTION
**Pull request by walkawayy**
#### Checklist

- [X] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description
Fixed some incorrectly rotated pickups when using the 3D pickups option. Shoutout @aredfan for the list.

### CODE REVIEW

**Comment by walkawayy** [comment]
@ajtudela requesting a translation please. 😃

**Comment by walkawayy** [comment]
Here's some examples of the fixed rotations:

Screenshot 2023-08-10 211357

Screenshot 2023-08-10 211409

Screenshot 2023-08-10 211646

Screenshot 2023-08-10 211739

Screenshot 2023-08-10 212000

Screenshot 2023-08-10 212037

Screenshot 2023-08-10 212131

Screenshot 2023-08-10 212611

**Comment by Richard-L** [comment]
Oh really nice 👍

**Comment by aredfan** [comment]
@walkawayy  What a nice surprise to see you're working on the rotate 3D pickups feature, thank you so much!

Glad to hear my list is being put to good use 👍

**Review by lahm86 — COMMENTED** [review]
If you have `walk_to_items` disabled, the pickups become shy and always face away from Lara.

https://github.com/LostArtefacts/Tomb1Main/assets/33758420/db294b00-7838-4bb8-9650-420e4dfefcc7


I think it might be caused by this:

https://github.com/LostArtefacts/Tomb1Main/blob/develop/src/game/objects/general/pickup.c#L89

In contrast with `Pickup_CollisionControlled`, it doesn't appear to restore the rotation after running through the checks. It seems to be the same in the `scion.c` collision functions as well.

**Comment by walkawayy** [comment]
Oops totally forgot to test with the option off. The original game ignored pickup rots because they were sprites that always faced Lara.

**Comment by ajtudela** [comment]
Done here: #915

**Comment by walkawayy** [comment]
Lmk if the `goto` is ok. Otherwise we need to invert all the conditions like `Pickup_CollisionControlled` to restore the item rots.

File: src/game/inject.c [hunk 1]
```diff
@@ -1484,6 +1500,30 @@ static void Inject_RoomDoorEdits(INJECTION *injection)
...
+    MYFILE *fp = injection->fp;
+
+    int16_t item_num;
+    int16_t y_rot;
+    for (int i = 0; i < inj_info->item_rotation_count; i++) {
+        File_Read(&item_num, sizeof(int16_t), 1, fp);
+        File_Read(&y_rot, sizeof(int16_t), 1, fp);
+
+        if (item_num < 0 || item_num >= g_LevelItemCount) {
+            LOG_WARNING("Item number %d is outwith level item range", item_num);
```
Comment (rr-) [inline src/game/inject.c]: outwith? :D

**Review by rr- — APPROVED** [review]

**Comment by lahm86** [comment]
> Lmk if the `goto` is ok. Otherwise we need to invert all the conditions like `Pickup_CollisionControlled` to restore the item rots.

I think `Scion_Collision` will need the same treatment as it currently rotates in Qualopec and Sanctuary. Atlantis/Great Pyramid are fine as they're animated items anyway; Pierre's drop uses `Pickup_Collision` so that's fine too.

File: src/game/inject.c (hunk 1 above)
Comment (walkawayy) [reply src/game/inject.c]: Lol Lahm helped out on this. Sorry to change native tongue, but I'll change it to make it easier to understand. 😢 😆

**Review by lahm86 — APPROVED** [review]

**Pull request merged**